### PR TITLE
Chris

### PR DESCRIPTION
**Pull request by ChrisC2020**
Did the backend for inserting training periods into the database. Also made some changes to the form for new training periods. You will need the database file which is not included. I will distribute it momentarily, import it into your local database through myphp.

### CODE REVIEW

File: selectTrainingPeriod.php [hunk 1]
```diff
@@ -1,5 +1,12 @@
 <?php
```
Comment (AlexKeehan) [inline selectTrainingPeriod.php:1]: Missing header comment for the file with name of file, description & author.

File: selectTrainingPeriod.php [hunk 2]
```diff
@@ -1,5 +1,12 @@
...
-    // Page for an Admin to select a Training Period.
+//require_once 'database\dbinfo.php'; //need to get this to point to dbinfo in the correct folder
+$con = mysqli_connect("localhost","duplicate","duplicate","duplicate");
```
Comment (AlexKeehan) [inline selectTrainingPeriod.php:3]: Workaround, but still in progress, so will be changed in future

File: selectTrainingPeriod.php [hunk 3]
```diff
@@ -21,7 +28,50 @@
         die();
     }
 
-    
+    //my code, comment out if not ready in time
+    if ($_SERVER["REQUEST_METHOD"] == "POST") {
+        //echo "hello world"; //okay this works for debugging
```
Comment (AlexKeehan) [inline selectTrainingPeriod.php:33]: Good debugging

File: selectTrainingPeriod.php [hunk 4]
```diff
@@ -21,7 +28,50 @@
...
+        $semester = $_POST["training-period"];
+        $year = $_POST["yeardropdown"];
+        //does not require error checking for adding to past years because the year dropdown is now dynamic
+        //may require error checking for adding training periods that've already passed
+        //but that could also be allowed, it wouldn't cause any problems to add them
+
```
Comment (AlexKeehan) [inline selectTrainingPeriod.php:39]: Good commenting

File: selectTrainingPeriod.php [hunk 5]
```diff
@@ -21,7 +28,50 @@
...
+                break;
+            case "Fall":
+                //echo "fall";
+                $startDate = "$year-09-01";
+                $endDate = "$year-10-31";
+                break;
+            default :
+                //echo "default";
+                break;
+            }
```
Comment (AlexKeehan) [inline selectTrainingPeriod.php:62]: Good use of switch statement instead of if statement

File: selectTrainingPeriod.php [hunk 6]
```diff
@@ -21,7 +28,50 @@
...
+
+        $query = "INSERT INTO `dbtrainingperiods` (`id`, `name`, `startdate`, `enddate`) VALUES (NULL, '$semester $year', '$startDate', '$endDate')";
+        try{
+            //echo "checkpoint";
+            $result= mysqli_query($con, $query);
+        } catch (Exception $e) {
```
Comment (AlexKeehan) [inline selectTrainingPeriod.php:68]: Good error catching

File: selectTrainingPeriod.php [hunk 7]
```diff
@@ -51,22 +101,43 @@
...
-                <option value="third">Sep 1 - Oct 31</option>
+                <select name="training-period">
+                <option value="Spring">Jan 1 - Feb 28</option>
+                <option value="Summer">May 1 - Jun 30</option>
+                <option value="Fall">Sep 1 - Oct 31</option>
                 </select>
             <h3>Select Year</h3>
+                <form method="post">
+                <select name="yeardropdown" id="yeardropdown">
+                <script>
```
Comment (AlexKeehan) [inline selectTrainingPeriod.php:119]: Good use of a script

File: selectTrainingPeriod.php [hunk 8]
```diff
@@ -21,7 +28,50 @@
...
+        //echo "hello world"; //okay this works for debugging
+        $semester = $_POST["training-period"];
+        $year = $_POST["yeardropdown"];
+        //does not require error checking for adding to past years because the year dropdown is now dynamic
+        //may require error checking for adding training periods that've already passed
+        //but that could also be allowed, it wouldn't cause any problems to add them
+
+        //calculating start and end dates given the semester and year values
+        //these values are hardcoded, we should redo this later to make it dynamic
+        //if we create relative dates for each semester and then use those in the form, it would only need changing 1 place
```
Comment (AlexKeehan) [inline selectTrainingPeriod.php:42]: Good commenting

File: selectTrainingPeriod.php [hunk 9]
```diff
@@ -21,7 +28,50 @@
...
+            }
+
+        $query = "INSERT INTO `dbtrainingperiods` (`id`, `name`, `startdate`, `enddate`) VALUES (NULL, '$semester $year', '$startDate', '$endDate')";
+        try{
+            //echo "checkpoint";
+            $result= mysqli_query($con, $query);
+        } catch (Exception $e) {
+            echo "training-period already present in database";
+        }
+        //echo "hello world";
```
Comment (AlexKeehan) [inline selectTrainingPeriod.php:71]: After training period is added to dbTrainingPeriods, reroute to addTrainingPeriod.php

**Review by AlexKeehan — COMMENTED** [review]
Pretty good commented and formatted code. Small changes are needed that are mentioned in the comments

File: selectTrainingPeriod.php (hunk 9 above)
Comment (ChrisC2020) [reply selectTrainingPeriod.php:71]: Will do!

**Pull request merged**